docs/conf.py

```diff
@@ -0,0 +1,74 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For a full list of options see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+
+# -- Path setup --------------------------------------------------------------
+
+from pathlib import Path
+import sys
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'The Operator Framework'
+copyright = '2020, Canonical Ltd'
+author = 'Canonical Ltd'
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx.ext.autodoc',
+    'sphinx.ext.napoleon',
+    'sphinx.ext.todo',
+    'sphinx.ext.viewcode',
+]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'nature' # 'alabaster'
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+
+
+# -- Options for sphinx.ext.todo ---------------------------------------------
+
+#  If this is True, todo and todolist produce output, else they
+#  produce nothing. The default is False.
+todo_include_todos = True
+
+# -- Options for sphinx.ext.autodoc ------------------------------------------
+
+# This value controls how to represents typehints. The setting takes the
+# following values:
+#     'signature' – Show typehints as its signature (default)
+#     'description' – Show typehints as content of function or method
+#     'none' – Do not show typehints
+autodoc_typehints = 'description'
+
```

Comment:
I missed that this doesn't have the autoclass_content = 'both' that you had me add before.

That needs to be added, but the rest look good to me. I'm sure there is more tweaking we can do over time, but having something is going to be a lot better than what we have now. :)

Reply:
I was waiting for your confirmation that that change had DTRT for you :slightly_smiling_face:  I'm taking this as a ‘yes’, and have pushed that change in.